app/src/main/kotlin/com/github/gotify/init/InitializationActivity.kt

```diff
@@ -106,6 +112,7 @@ internal class InitializationActivity : AppCompatActivity() {
 
         settings.setUser(user.name, user.isAdmin)
         requestVersion {
+            splashScreenActive = false
```

Comment:
This Closure in requestVersion doesn't run on the main thread, could there be thread safety problems with the splashScreenActive variable?

Reply:
I'm not sure there.
As long as that variable is false, the splashScreen will stay active.
And as long as the variable at some point jumps to true, the splashScreen will disappear.

If there could be cases that the variable does not get changed, tbh I don't know.

Reply:
Am I stupid or do we already run 'callInUI' in requestVersion?

---

![image](https://user-images.githubusercontent.com/46033730/216326998-535bfcba-7c93-4497-84bc-5a1046da2e35.png)

Reply:
Uff, sorry, I was pretty sure I only saw Callback.call there (:.

Reply:
Alright, no problem